Extensions/xkit_preferences.js

```diff
@@ -394,7 +420,7 @@ XKit.extensions.xkit_preferences = new Object({
 
 		},
 
-		check: function(id) {
+		check: function(id, version) {
```

Comment:
When I spoke to you on [Gitter](https://gitter.im/new-xkit/XKit?at=56a0c178da358920486ffd92), you mentioned that the purpose of `check` is to see if you need to save or modify an item. It might be more meaningful to rename this function into something like `shouldSave`.

Also, `check` is returning three different values; it might make sense to split this into two different functions.